EventListener/ControllerListener.php

```diff
@@ -50,13 +50,18 @@ public function __construct(Reader $reader)
      */
     public function onKernelController(FilterControllerEvent $event)
     {
-        if (!is_array($controller = $event->getController())) {
+        if (!is_array($controller = $event->getController()) && !is_object($controller)) {
```

Comment:
In case of anonymous functions (I think that's what @stof is aiming at), they cannot have Annotations and should therefore be skipped in this listener. I think that the only format that should be supported is `[controller, action]` where `action` can be `*Action` or `__invoke`. The first is not mandatory (I think) but is a requirement for auto-guessing template names.